include/core/rtos/Mutex.hpp

```diff
@@ -0,0 +1,120 @@
+#ifndef EVT_RTOS_MUTEX_
+#define EVT_RTOS_MUTEX_
+
+#include <core/rtos/Initializable.hpp>
+
+namespace core::rtos {
+
+/**
+ * Class that wraps a Threadx Mutex. Mutexes are used to create mutual exclusion between threads.\n\n
```

Comment:
You don't have to accept all my suggestions, but I'm going to go crazy if you leave `\n\n` everywhere

Reply:
The \n\n make the IDE format things nicer for the method documentation. Just read past it.

Reply:
For my own edification, what does this have to do with the IDE? Is this just CLion, or other IDEs?

I also would tend to want to avoid doing things differently in one PR from the rest of EVT-core. I feel like this is just going to end up being different all across the codebase, and I find that frustrating. If having `\n\n` is objectively better and we do it everywhere from here on out, I'm okay with it, but otherwise, I would prefer to defer to how things are until we have the time to update everything at once

Reply:
I only have knowledge of CLion, I do not know if other IDEs work the same way.
Here's an example:
With \n\n:
![image](https://github.com/user-attachments/assets/67527377-e90e-4c67-b601-c8b7118e292b)
Without \n\n:
![image](https://github.com/user-attachments/assets/cceb53ff-2e5e-4b05-a10a-bfddfde45f64)

You can see that the \n\n essentially spaces out the IDE on-hover info panel into two blocks, which partitions the information nicely. For me, it means that the information someone REALLY needs is right there for them to read, and sort of boilerplate information that they _might_ need but they don't necessarily is there for them if they want to read it, but it's visually separated from the other information.